### DISCUSSION TITLE
torch_geometric expand?

### DISCUSSION THREAD

**Discussion by ldv1** [pyg-team/pytorch_geometric#8062] [Q&A] [answered]
Hi,

After a global pooling operation, I would like to "expand" the result to the graphs in the batch. Stated otherwise, I need the opposite of scatter. To illustrate, something like `torch_geometric.XYZ.expand` in

```
def forward(self, x, batch):
    global_max = global_max_pool(x, batch=batch)        
    y = torch_geometric.XYZ.expand(global_max, batch=batch)
    return x*y
```

In this case, the global max vector for a given graph is simply repeated for each node of the graph.

Is such a function already available?
Thanks for any help.

**Comment by rusty1s** [comment, marked as answer]
You can just do
```python
global_max = global_max_pool(x, batch=batch)        
y = global_max[batch]

**Reply by ldv1** [reply]
Many thanks!
